src/core/useAtom.ts

```diff
@@ -61,8 +61,9 @@ export function useAtom<Value, Update>(
   )
 
   const StoreContext = getStoreContext(atom.scope)
-  const [mutableSource, updateAtom] = useContext(StoreContext)
+  const [mutableSource, updateAtom, commit] = useContext(StoreContext)
   const value: Value = useMutableSource(mutableSource, getAtomValue, subscribe)
+  useEffect(() => commit())
```

Comment:
What if it's unmounting, while calling commit? You might need a guard here.

Reply:
if this atom is unmouting, it doesn't do anything. I'm not sure if I get your question.
btw, we should explicitly ignore the return value of commit here.

Reply:
If the effect runs while it's unmounting, it will throw something like this: 
```
Warning: Can't perform a React state update on an unmounted component.
This is a no-op, but it indicates a memory leak in your application.
```

Reply:
No worries about it. uMS handles it already.